zenoh-ext/examples/z_nbftr_pub.rs

```diff
@@ -0,0 +1,113 @@
+//
+// Copyright (c) 2022 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+use async_std::task::sleep;
+use clap::{App, Arg};
+use std::time::Duration;
+use zenoh::config::Config;
+use zenoh::prelude::r#async::*;
+use zenoh_ext::*;
+
+#[async_std::main]
+async fn main() {
+    // Initiate logging
+    env_logger::init();
+
+    let (config, key_expr, value, cache, history) = parse_args();
+
+    println!("Opening session...");
+    let session = zenoh::open(config).res().await.unwrap();
+
+    println!("Declaring NBFTReliablePublisher on {}", &key_expr);
+    let publ = session
+        .declare_reliable_publisher(&key_expr)
+        .with_cache(cache)
```

Comment:
A cache associated to a publisher will be preconfigured for it. It will typically have the publisher id as queryable prefix. So taking a cache declared independently does not seem the best way to me. Still a better name for the function could be found...